composites/SnippetPreview/components/Icon.js

```diff
@@ -0,0 +1,34 @@
+import React from "react";
+import PropTypes from "prop-types";
+import styled from "styled-components";
+
+import * as Icons from "../../../style-guide/svg";
+
+/**
+ * Returns an icon.
+ *
+ * @param {object} props Component props.
+ *
+ * @returns {ReactElement} Icon.
+ */
+export const Icon = ( props ) => {
+	const IconComponent = styled( Icons[ props.icon ] )`
```

Comment:
I think it might be better to pass the actual icon in with the props. Now `props.icon` is a string which means that it is hard to see where a certain icon has been used.

If you do it like this, it is clearer:
```
import edit from "[path-to-icons]/edit.svg";

// (...)

renderFunc() {
     return <Icon icon={ edit } />;
}
```